### PR TITLE
CI: remove AppVeyor, add Windows and macOS to CI testing, remove 'dev'

### PR DESCRIPTION
**Pull request by mwtoews**
This is for the `maint-1.8` branch. I'll leave a few inline comments...

### CODE REVIEW

File: .github/workflows/tests.yml [hunk 1]
```diff
@@ -36,15 +42,15 @@ jobs:
             speedups: 0
           # 2019
           - python: 3.8
-            geos: 3.8.2
+            geos: 3.8.1
```
Comment (mwtoews) [inline .github/workflows/tests.yml:45]: downgrade is necessary to compile GEOS for Windows

File: .github/workflows/tests.yml [hunk 2]
```diff
@@ -120,42 +145,62 @@ jobs:
...
       - name: Overview of the Python environment (pip list)
-        shell: bash
         run: |
           pip list
 
       - name: Run tests
-        shell: bash
-        continue-on-error: ${{ matrix.geos == 'main' }}
         run: |
+          export DYLD_LIBRARY_PATH=${{ env.DYLD_LIBRARY_PATH }}
```
Comment (mwtoews) [inline .github/workflows/tests.yml:189]: this appears to be necessary! Despite being part of `$GITHUB_ENV`, it cannot be seen i.e. `echo "DYLD_LIBRARY_PATH=$DYLD_LIBRARY_PATH"` shows a blank entry.

File: ci/install_geos.sh [hunk 3]
```diff
@@ -1,10 +1,11 @@
-#!/bin/sh
+#!/bin/bash
```
Comment (mwtoews) [inline ci/install_geos.sh:1]: pushd / popd are bash-specific

File: .github/workflows/tests.yml [hunk 4]
```diff
@@ -70,46 +76,65 @@ jobs:
           - python: "3.10"
             geos: 3.10.1
             speedups: 0
-          # dev
-          - python: "3.10"
-            geos: main
-            speedups: 1
```
Comment (mwtoews) [inline .github/workflows/tests.yml:76]: given that this is a maintenance branch, it does not need to test against the latest features.

**Review by jorisvandenbossche — APPROVED** [review]
Thanks!

I suppose the changes to install script are just a copy from the main branch?

**Comment by mwtoews** [comment]
yes, except for one difference
```diff
1c1
< #!/bin/sh
---
> #!/bin/bash
```
because pushd and popd are bash specific.

**Pull request merged**